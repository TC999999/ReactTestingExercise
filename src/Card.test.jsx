import { render } from "@testing-library/react";
import Card from "./Card.jsx";

it("renders without crashing", function () {
  render(
    <Card caption="test-caption" src="not-a-link" currNum={1} totalNum={100} />
  );
});

it("should match snapshot", () => {
  const { asFragment } = render(
    <Card caption="test-caption" src="not-a-link" currNum={1} totalNum={100} />
  );
  expect(asFragment()).toMatchSnapshot();
});
