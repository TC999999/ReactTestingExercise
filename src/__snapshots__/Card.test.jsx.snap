// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test-caption
    </h4>
    <img
      alt="test-caption"
      class="Card-image"
      src="not-a-link"
    />
    <small
      class="Card-small"
    >
      Image 1 of 100.
    </small>
  </div>
</DocumentFragment>
`;
